sdk/textanalytics/Azure.AI.TextAnalytics/api/Azure.AI.TextAnalytics.netstandard2.0.cs

```diff
@@ -423,6 +441,18 @@ public partial class DocumentSentiment
         public Azure.AI.TextAnalytics.TextSentiment Sentiment { get { throw null; } }
         public System.Collections.Generic.IReadOnlyCollection<Azure.AI.TextAnalytics.TextAnalyticsWarning> Warnings { get { throw null; } }
     }
+    public partial class DynamicClassifyDocumentResultCollection : System.Collections.ObjectModel.ReadOnlyCollection<Azure.AI.TextAnalytics.ClassifyDocumentResult>
+    {
+        internal DynamicClassifyDocumentResultCollection() : base (default(System.Collections.Generic.IList<Azure.AI.TextAnalytics.ClassifyDocumentResult>)) { }
+        public string ModelVersion { get { throw null; } }
+        public Azure.AI.TextAnalytics.TextDocumentBatchStatistics Statistics { get { throw null; } }
+    }
+    public partial class DynamicClassifyOptions : Azure.AI.TextAnalytics.TextAnalyticsRequestOptions
+    {
+        public DynamicClassifyOptions(System.Collections.Generic.IList<string> categories) { }
```

Comment:
I passed it as a constructor parameter because `Categories` is a required property, but see my following comment because I think there's a better way and want to know what you think: https://github.com/Azure/azure-sdk-for-net/pull/32593#discussion_r1028375249

Reply:
What's not clear is whether the property should be read-only henceforth. We don't want customers clearing it before sending if it's required (by default we should not typically be serializing an empty collection). `ICollection<T>` defines `Clear`, so `IList<T>` would have it to. I would talk with @KrzysztofCwalina and @tg-msft about this. Perhaps if it's passed in as a parameter, backing properties should be read-only which means customers can't mutate it after. If anything, that is equivalent to other constructor parameters now that should be read-only (i.e. no `set` accessor) for those backing constructor parameters.